### PR TITLE
INS-25749 Support Cadence HTTP API option

### PR DESCRIPTION
**Pull request by justinweng-instaclustr**
- Add the boolean option `use_http_api` for provisioning Cadence.
- Add the firewall rule type `CADENCE_HTTP_API` to allow remote connection to Cadence HTTP API endpoints.

### CODE REVIEW

File: docs/resources/cadence_cluster_v2.md [hunk 1]
```diff
@@ -64,6 +65,9 @@ The following terms are used to describe attributes in the schema of this resour
 *___sla_tier___*<br>
 <ins>Type</ins>: string, required, immutable<br>
 <ins>Constraints</ins>: allowed values: [ `PRODUCTION`, `NON_PRODUCTION` ]<br><br>SLA Tier of the cluster. Non-production clusters may receive lower priority support and reduced SLAs. Production tier is not available when using Developer class nodes. See [SLA Tier](https://www.instaclustr.com/support/documentation/useful-information/sla-tier/) for more information.<br><br>
+*___use_http_api___*<br>
+<ins>Type</ins>: boolean, required, immutable<br>
```
Comment (ronycsdu) [inline docs/resources/cadence_cluster_v2.md:69]: Should this be a required setting? Or optional and if absent we treat as false?
Check with #team-customerplatform about what should be the best practice.

Comment (justinweng-instaclustr) [reply docs/resources/cadence_cluster_v2.md:69]: I'll check with them

Comment (iclonghoang) [reply docs/resources/cadence_cluster_v2.md:69]: this is better be optional with default false

adding new `required` property means breaking compatibility and we'll need to bump the API endpoint version

**Comment by iclonghoang** [comment]
LGTM
pending Makefile version bump, which can be done before prod release

**Comment by justinweng-instaclustr** [comment]
> LGTM pending Makefile version bump, which can be done before prod release

Cool, thanks. I'll bump the version before prod release.

**Review by iclonghoang — APPROVED** [review]

**Review by ronycsdu — APPROVED** [review]

**Pull request merged**